pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/engine/PtNDArray.java

```diff
@@ -54,8 +54,12 @@
     private PtNDManager manager;
     private PtNDArrayEx ptNDArrayEx;
 
+    // keep a reference to direct buffer to avoid GC release the memory
+    private ByteBuffer dataRef;
```

Comment:
if we close the NDArray, and not shut down the dataRef. Will the dataRef (pointer to native memory) still exist and waiting for GC to clean up?

Reply:
Yes.
1. DirectBuffer has a different lifecycle than normal GC, it doesn't use heap
2. We are creating DirectBuffer before, now just holding it a bit longer.

Reply:
when we close the NDArray, we should mark the DirectByteBuffer to be ready for GC. As long as Java hold the reference, the data would be still there.

Reply:
DirectBuffer are using Phantom reference instead. It will be collected with cleaner thread instead of gc. Should be safe then...